src/shared/version_op.c

```diff
@@ -661,11 +661,26 @@ os_info *get_unix_version()
             } else if(strcmp(strtok_r(buff, "\n", &save_ptr),"Darwin") == 0){
                 info->os_platform = strdup("darwin");
 
-                if (cmd_output_ver = popen("sw_vers -productName", "r"), cmd_output_ver) {
-                    if(fgets(buff, sizeof(buff) - 1, cmd_output_ver) == NULL){
-                        mdebug1("Cannot read from command output (sw_vers -productName).");
-                    } else {
-                        w_strdup(strtok_r(buff, "\n", &save_ptr), info->os_name);
+                if (cmd_output_ver = popen("system_profiler SPSoftwareDataType", "r"), cmd_output_ver)
+                {
+                    if (fread(buff, 1, sizeof(buff), cmd_output_ver) > 0)
```

Comment:
fread is not string-safe, and magic numbers should be avoided. If read information is bigger than the buffer this could lead to a buffer overrun.